client/client.go

```diff
@@ -266,20 +266,20 @@ type serviceModeKeeper struct {
 	// triggering service mode switching concurrently.
 	sync.RWMutex
 	serviceMode     pdpb.ServiceMode
-	tsoClient       atomic.Value // *tsoClient
+	tsoClient       *tsoClient
 	tsoSvcDiscovery ServiceDiscovery
 }
 
-func (smk *serviceModeKeeper) close() {
-	smk.Lock()
-	defer smk.Unlock()
-	switch smk.serviceMode {
+func (k *serviceModeKeeper) close() {
+	k.Lock()
```

Comment:
Rlock is enough?

Reply:
It should be safer to use lock, shouldn't it? If we use RLock, even after the tsoServiceDiscovery and tsoClient have started the Close(), it will still allow access to tsoServiceDiscovery and tsoClient internal unnecessarily.

Reply:
ok